lib/auth/auth_with_roles.go

```diff
@@ -1126,6 +1126,15 @@ func (a *ServerWithRoles) PingInventory(ctx context.Context, req proto.Inventory
 	return a.authServer.PingInventory(ctx, req)
 }
 
+func (a *ServerWithRoles) UpdateLabels(ctx context.Context, req proto.InventoryUpdateLabelsRequest) error {
+	// admin-only for now, but we'll eventually want to develop an RBAC syntax for
+	// the inventory APIs once they are more developed.
+	if !a.hasBuiltinRole(types.RoleAdmin) {
```

Comment:
Yes, we have a Q2 goal to remove all `RoleAdmin` checks.

Reply:
@zmb3 @rosstimothy Yeah we have this goal but do we actually know what we want to replace it with? What other way do we have to make sure that certain action can only be run on auth server? For example, we've also been planning to only allow RoleAdmin for the new [API key resource](https://github.com/gravitational/teleport/pull/26599) that @mdwn is adding as a part of the hosted plugins work.

Reply:
If the goal is to make things work for Cloud customers, I think it should be ok to keep RoleAdmin checks for cases/resources which users aren't expected to modify. I don't know if we can just commit to removing all RoleAdmin checks unconditionally.

Reply:
Right, if it's something that cloud users should never see (like when we want to control API keys for them), then these `RoleAdmin` checks are okay and don't need to be removed.

This should be rare though, in most cases we don't want features that require direct auth access since it doesn't work on cloud. Inventory commands are something cloud customers would expect to work, so we want to replace it with actual RBAC checks (as noted in the comment).

Reply:
Now that I think about it, the only thing that needs to be able to call this is the eventual reconciliation service on the auth servers. I don't think this needs to be implemented for `tctl` at all, since the desired behavior can be accomplished by updating `ServerInfo` resources.